lib/database/model/sthDatabaseNameCodecTool.js

```diff
@@ -383,34 +383,89 @@ function encodeOrDecodeCollection(options, databaseName, collectionName, callbac
+    db.command(listCommand, function onListCollections(err, results) {
         if (err) {
             return process.nextTick(callback.bind(null, err));
         }
-        sthDatabase.connection.db(databaseName).dropDatabase(function onDropDatabase(err) {
-            if (err) {
-                return process.nextTick(callback.bind(null, err));
-            }
-            sthLogger.info(
-                sthConfig.LOGGING_CONTEXT.DB_LOG,
-                // prettier-ignore
-                (options.encode ? 'Codification' : 'Decodification') + " of database '" + databaseName +
-                    ' successfully completed.'
-            );
-            process.nextTick(callback);
+
+        var arrayCollectionRenames = [];
+        results.cursor.firstBatch.forEach((collection) => {
+            var source = sourceDB + '.' + collection.name;
+            var destination = targetDB + '.' + collection.name;
+
+            let collectionData = { source: source, destination: destination, db: adminDB };
+            arrayCollectionRenames.push(collectionData);
         });
+
+        async.map(
+            arrayCollectionRenames,
+            function(element, callback) {
+                element.db.command(
+                    { renameCollection: element.source, to: element.destination, dropTarget: true },
+                    function(err) {
+                        callback(err);
+                    }
+                );
+            },
+            function(err) {
+                if (err) {
+                    return process.nextTick(callback.bind(null, err));
+                }
+                sthLogger.info(
+                    sthConfig.LOGGING_CONTEXT.DB_LOG,
+                    // prettier-ignore
+                    (options.encode ? 'Codification' : 'Decodification') + " of database '" + databaseName +
+                        ' successfully completed.'
+                );
+                process.nextTick(callback);
+            }
+        );
     });
 }
 
+/**
+ *  Move collections from one Database to another, deleting previous DB
+ * 
+ * @param {Mongo database}  db 
+ * @param {string}          sourceDB 
+ * @param {string}          targetDB 
+ * @param {boolean}         dropSource 
+ * @param {callback}        callback 
+ 
+var cloneDatabse = function (db, sourceDB, targetDB, dropSource, callback) {
+    listCollections(db.db(sourceDB), function(results){
+        results.forEach(collection => {
+            var source = sourceDB + "." + collection.name;
+            var destination = targetDB + "." + collection.name;
+            db.command({'renameCollection': source, 
+                        'to': destination, 
+                        'dropTarget': true
+                        },
+                        function(err, results) {
+                        });
+            // No needed to drop DB
+                        
+        });
+        callback();
+    });
+
+};
+*/
```

Comment:
Remove leftover block.

Reply:
Fixed in https://github.com/telefonicaid/fiware-sth-comet/pull/554/commits/4a63305fb60a5e2a0a87738555b684b9717488b9